CoreVisualGraph/src/au/gov/asd/tac/constellation/graph/visual/plugins/blaze/BlazeActions.java

```diff
@@ -113,10 +194,217 @@ public BlazeActions() {
         menu.addChangeListener(e -> {
             if (graph != null) {
                 updateSliders(graph);
+                updateCustomBlazes();
             }
         });
         menu.setEnabled(false);
 
+        colorPanel1 = new JPanel();
+        colorPanel2 = new JPanel();
+        colorPanel3 = new JPanel();
+        colorPanel4 = new JPanel();
+        colorPanel5 = new JPanel();
+        colorPanel6 = new JPanel();
+        colorPanel7 = new JPanel();
+        colorPanel8 = new JPanel();
+        colorPanel9 = new JPanel();
+        colorPanel10 = new JPanel();
+
+        colorPanels = new ArrayList<>();
+        colorPanels.add(colorPanel1);
+        colorPanels.add(colorPanel2);
+        colorPanels.add(colorPanel3);
+        colorPanels.add(colorPanel4);
+        colorPanels.add(colorPanel5);
+        colorPanels.add(colorPanel6);
+        colorPanels.add(colorPanel7);
+        colorPanels.add(colorPanel8);
+        colorPanels.add(colorPanel9);
+        colorPanels.add(colorPanel10);
+
+        final List<ConstellationColor> colors = getPresetCustomColors();
+
+        if (colors != null) {
+            int panelCounter = 0;
+            for (final JPanel panel : colorPanels) {
+                if (panelCounter < colors.size()) {
+                    if (colors.get(panelCounter) != null) {
+                        panel.setBackground(colors.get(panelCounter++).getJavaColor());
+                    }
+                }
+            }
+        }
+
+        final JPanel verticalPanel = new JPanel();
+        verticalPanel.setMaximumSize(new Dimension(200, 75));
+
+        final GridLayout gridLayout = new GridLayout(0, 5, 5, 5);
+        verticalPanel.setLayout(gridLayout);
+        verticalPanel.setBorder(BorderFactory.createTitledBorder("Preset Colors"));
+
+        for (final JPanel colorPanel : colorPanels) {
+            colorPanel.setBackground(DEFAULT_COLOR);
+            verticalPanel.add(colorPanel);
+        }
+
+        colorPanel1.setMaximumSize(new Dimension(5, 5));
+        colorPanel1.addMouseListener(new MouseAdapter() {
```

Comment:
can this be a lambda expression (or a method reference)? If so, can you do it for all the colour panels?

Reply:
I couldn't make it a lambda because it wasn't a functional interface, so I made it call the method.